scrapi/harvesters/push_api.py

```diff
@@ -0,0 +1,108 @@
+import six
+import json
+import logging
+from datetime import date
+from datetime import timedelta
+
+import requests
+
+from scrapi import settings
+from scrapi.base import BaseHarvester, HarvesterMeta
+from scrapi.linter.document import RawDocument, NormalizedDocument
+
+logger = logging.getLogger(__name__)
+
+
+def gen_harvesters():
+    try:
+        return {
+            source['shortname']: gen_harvester(**source)
+            for source in get_sources()
+        }
+    except Exception as e:
+        logger.warn('Could not generate harvesters from push api')
+        if settings.DEBUG:
+            logger.exception(e)
+
+
+def get_sources():
+    response = requests.get('{}/provider_list'.format(settings.SHARE_REG_URL)).json()
+    while True:
+        for source in response['results']:
+            yield source
+        if response['next']:
+            response = requests.get(response['next']).json()
+        else:
+            break
+
+
+def gen_harvester(shortname=None, longname=None, url=None, favicon_dataurl=None, **kwargs):
+    assert shortname and longname and url and favicon_dataurl
+    logger.info('Generating harvester {}'.format(shortname))
+
+    from scrapi.processing.elasticsearch import DatabaseManager
```

Comment:
Does this make having elasticsearch setup mandatory? What happens if we remove elasticsearch in the future?